iroha/src/sumeragi.rs

```diff
@@ -1,14 +1,216 @@
-use crate::prelude::*;
+use crate::{
+    crypto::{Crypto, Hash},
+    peer::{self, PeerId},
+    prelude::*,
+};
+use iroha_derive::*;
+use parity_scale_codec::{Decode, Encode};
+use std::cmp::Ordering;
 
-#[derive(Default)]
-pub struct Sumeragi {}
+pub struct Sumeragi {
```

Comment:
Probably need to add state info about how many failures there were in the current round (index) to know the current order of the nodes after a leader election.